Configuration for TypeScript Create-React-App

First of all, thanks for sharing this great tool with us.

I used `Poetic` with a new TypeScript Create-React-App and needed to do some configurations to eliminate the VSCode **IDE** warnings.

I will share the issues I encountered and my solutions for other users and maybe for the author to incorporate them in the project.

 ## Create React App with TypeScript

- [CRA: Adding TypeScript](https://create-react-app.dev/docs/adding-typescript)

```
npx create-react-app react-ts --typescript
cd react-ts
npm install --save typescript @types/node @types/react @types/react-dom @types/jest
```

## `Poetic`

```
npx poetic
```

- I have the following VSCode extensions installed
	- ESLint
	- Prettier
	- EditorConfig

## Warnings

### JSX not allowed in files with extension '.tsx' eslint(react/jsx-filename-extension)

- Add the following rule in `.eslintrc.js`

```
  rules: {
    'react/jsx-filename-extension': [1, { extensions: ['.jsx','.tsx'] }],
  },
```

### ESLintIgnore `serviceWorker.ts`

- Add the following to `.eslintignore`
  - `src/serviceWorker.ts`

### Cannot find module './logo.svg'. ts(2307)

- Add the following to `src/react-app-env.d.ts`

```
// declare webpack modules
declare module '*.png';
declare module '*.svg';
declare module '*.jpg';
declare module '*.gif';
declare module '*.less';
```

--- a/boilerplate/.eslintignore
+++ b/boilerplate/.eslintignore
@@ -1,1 +1,1 @@
-serviceWorker.js
+serviceWorker.*

--- a/boilerplate/tsconfig.json
+++ b/boilerplate/tsconfig.json
@@ -1,10 +1,25 @@
 {
-  /* Poetic base configuration */
+  /* Poetic base style configuration */
   "extends": "./node_modules/poetic/config/typescript/tsconfig.json",
   /* Add your custom configuration after this */
-  "target": "es5",
   "include": ["src/**/*"],
   "compilerOptions": {
-    "outDir": "dist"
+    "target": "es5",
+    "outDir": "dist",
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
+    ],
+    "skipLibCheck": true,
+    "esModuleInterop": true,
+    "module": "esnext",
+    "moduleResolution": "node",
+    "isolatedModules": true,
+    "resolveJsonModule": true,
+    "allowSyntheticDefaultImports": true,
+    "allowJs": true,
+    "jsx": "react",
+    "noEmit": true
   }
 }

--- a/config/eslint/eslint-config.js
+++ b/config/eslint/eslint-config.js
@@ -31,7 +31,8 @@ module.exports = {
   },
   rules: {
     quotes: ["warn", "single", { "allowTemplateLiterals": true }],
-    "no-unexpected-multiline": "error", // https://bit.ly/2kMIX99
+    "no-unexpected-multiline": "error", // https://bit.ly/2kMIX99,
+    "react/jsx-filename-extension": [1, { "extensions": [".jsx", ".tsx"] }],
   },
   overrides: [
     {

--- a/config/typescript/tsconfig.json
+++ b/config/typescript/tsconfig.json
@@ -1,19 +1,23 @@
 {
   "compilerOptions": {
+    
     /* Strict Type-Checking Options */
-    "strict": false,                       /* Enable all strict type-checking options. */
-    "noImplicitAny": true,                 /* Raise error on expressions and declarations with an implied 'any' type. */
-    "strictNullChecks": true,              /* Enable strict null checks. */
-    "strictFunctionTypes": true,           /* Enable strict checking of function types. */
-    "strictBindCallApply": true,           /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
-    "strictPropertyInitialization": true,  /* Enable strict checking of property initialization in classes. */
-    "noImplicitThis": true,                /* Raise error on 'this' expressions with an implied 'any' type. */
-    "alwaysStrict": true,                  /* Parse in strict mode and emit "use strict" for each source file. */
+    "strict": false,                         /* Enable all strict type-checking options. */
+    "noImplicitAny": true,                   /* Raise error on expressions and declarations with an implied 'any' type. */
+    "strictNullChecks": true,                /* Enable strict null checks. */
+    "strictFunctionTypes": true,             /* Enable strict checking of function types. */
+    "strictBindCallApply": true,             /* Enable strict 'bind', 'call', and 'apply' methods on functions. */
+    "strictPropertyInitialization": true,    /* Enable strict checking of property initialization in classes. */
+    "noImplicitThis": true,                  /* Raise error on 'this' expressions with an implied 'any' type. */
+    "alwaysStrict": true,                    /* Parse in strict mode and emit "use strict" for each source file. */
 
     /* Additional Checks */
-    "noUnusedLocals": true,                /* Report errors on unused locals. */
-    "noUnusedParameters": true,            /* Report errors on unused parameters. */
-    "noImplicitReturns": true,             /* Report error when not all code paths in function return a value. */
-    "noFallthroughCasesInSwitch": true     /* Report errors for fallthrough cases in switch statement. */
+    "noUnusedLocals": true,                  /* Report errors on unused locals. */
+    "noUnusedParameters": true,              /* Report errors on unused parameters. */
+    "noImplicitReturns": true,               /* Report error when not all code paths in function return a value. */
+    "noFallthroughCasesInSwitch": true,      /* Report errors for fallthrough cases in switch statement. */
+    "forceConsistentCasingInFileNames": true /* Disallow inconsistently-cased references to the same file. */
   }
 }
+
+

